analyses/cell-type-wilms-tumor-06/notebook/07_combined_annotation_across_samples_exploration.Rmd

```diff
@@ -398,121 +541,499 @@ cell_type_df <- cell_type_df |>
+                   "has_loss_chr10p",
+                   "has_loss_chr10q",
+                   
+                   "has_loss_chr11p",
+                   "has_loss_chr11q",
+                   
+                   "has_loss_chr12p",
+                   "has_loss_chr12q",
+                   
+                   "has_loss_chr13p",
+                   "has_loss_chr13q",
+                   
+                   "has_loss_chr14p",
+                   "has_loss_chr14q",
+                   
+                   "has_loss_chr15p",
+                   "has_loss_chr15q",
+                   
+                   "has_loss_chr16p",
+                   "has_loss_chr16q",
+                   
+                   "has_loss_chr17p",
+                   "has_loss_chr17q",
+                   
+                   "has_loss_chr18p",
+                   "has_loss_chr18q",
+                   
+                   "has_loss_chr19p",
+                   "has_loss_chr19q",
+                   
+                   "has_loss_chr20p",
+                   "has_loss_chr20q",
+                   
+                   "has_loss_chr21p",
+                   "has_loss_chr21q",
+                   
+                   "has_loss_chr22p",
+                   "has_loss_chr22q"
+                   )
+
+
+#################################################################################
+# Here we define the size of each chromosome arm that will be used for plotting of the mean CNV profile
+
+  # Load cytoBand file
+  cytoBand <- read.table(gzfile(arm_order_file), header = FALSE, sep = "\t", stringsAsFactors = FALSE)
+  colnames(cytoBand) <- c("chrom", "start", "end", "band", "stain")
+  
+  # Extract chromosome arm information
+  cytoBand <- cytoBand |>
+    mutate(arm = substr(band, 1, 1))  # Extract 'p' or 'q' from the band column
+  
+  # Define arm boundaries
+  chromosome_arms <- cytoBand |>
+    group_by(chrom, arm) |>
+    summarize(
+      Start = min(start),
+      End = max(end),
+      .groups = "drop"
+    ) |>
+    mutate(Size = End - Start,
+           dupli = paste0("has_dupli_", chrom, arm),
+           loss = paste0("has_loss_", chrom, arm)) |> 
+  filter(arm != "")
+  
+  chromosome_arms <- chromosome_arms |>
+    mutate(Size = Size / sum(Size))
+
+  
+```
+
+```{r fig.width=20, fig.height=5, out.width='100%', results='asis'}
+
+# prepare the table for the number of sample with a (sub)clonal loss
+
+CNV_profile_loss <- cell_type_df_sub |>
+  select(sample_id, contains("has_loss_chr")) |>
+  group_by(sample_id) |>
+  summarise_all(.funs = mean) |>
+  pivot_longer(cols = -sample_id) |>
+  rename("chromosome_arm" = "name", "percentage" = "value") |>
+  left_join( chromosome_arms, by = c("chromosome_arm"="loss")) |>
+  mutate(chromosome_arm = factor(chromosome_arm, levels = has_loss_chr)) |>
+  mutate(absolute_70 = case_when((percentage > 0.7) ~ -1,
+         .default = 0),
+         absolute_10 = case_when((percentage > 0.1) ~ -1,
+         .default = 0)) 
```

Comment:
I think you mean `ifelse` here? You only have two options:

```suggestion
  mutate(
    absolute_70 = ifelse(percentage > 0.7, 1, 0), 
    absolute_10 = ifelse(percentage > 0.1, 1, 0)
 )
```

Reply:
Also, can you please add some text explaining this? These end up as different colors in the plot, but there is no legend or discussion of how to interpret it.